torch/testing/_internal/common_methods_invocations.py

```diff
@@ -145,6 +145,104 @@ def _getattr_qual(obj, name, default=_NOTHING):
         else:
             raise
 
+# NOTE: OpInfos and the OpInfo testing pattern
+# See also: https://github.com/pytorch/pytorch/wiki/Writing-tests-in-PyTorch-1.8
+# See also: https://github.com/pytorch/pytorch/issues/54261, which tracks
+#   which OpInfos we still want to create
+#
+# An "OpInfo", described by the class and subclasses below, is a collection
+#   of metadata related to a PyTorch operator, like torch.add. This metadata
+#   can be separated into three categories:
+#
+#     1) Metadata describing the operator, like "dtypesIfCPU", which describes
+#          the dtypes the operator supports on the CPU. This metadata
+#          is used by generated tests to understand what they should test
+#          and what the expected results of the test should be.
+#     2) Test directives, like "skips", which indicates that some generated
+#          tests should be skipped for this operator. Test directives
+#          directly control the behavior of generated tests.
+#     3) A "sample_inputs_func", a function that can generate valid inputs to
+#          the operator. Generating valid inputs is important for "black box"
+#          testing, like PyTorch's autograd tests, which can infer if an
+#          operator is behaving correctly just by observing how changes in
+#          to an operator's inputs affect its gradient computation.
+#
+# Most OpInfos are collected in the "op_db" sequence, although some specialized
+#   tests have their own OpInfo-based datastructures.
+#
+# OpInfos are intended to replace custom test generators, like "method_tests()"
+#   (see below) and the "tensor_op_tests" in test_torch.py. These generators
+#   proved hard to read and maintain, and having multiple operator lists
+#   was frequently confusing, with engineers not understanding what each
+#   list actually did. OpInfos hopes to address these issues and extend
+#   automated test coverage by consolidating these lists into one structured and
+#   documented list.
+#
+# Most tests that use every OpInfo are in test_ops.py. More correctly,
+#   test_ops.py contains "test templates" which are instantiated by
+#   PyTorch's "device generic test framework". See the developer Wiki article
+#   above for more information on this test instantiation process.
+#
+# In addition to the OpInfo base class, there are subclasses of OpInfos, like
```

Comment:
It might be useful to list all the `OpInfo` subclasses in master at the time of writing, to show the breadth of operator coverage.